### ISSUE TITLE
False positives for "colons": "always" when nesting

### ISSUE THREAD

**Issue by futekov** [rossPatton/stylint#181]
I discovered several false positives for `"colons": "always"` when nesting, this code produces 5 errors - one for each root class:

``` stylus
.ExtendBug
  @extend .List
  margin: 15px

.ParentReferenceBug
  max-width: 100px
  .ie9 &
    max-width: 80px

.ChildBug1
  .parent > .child
    float: left

.ChildBug2
  > .child
    padding: 4px

.ChildBug3
  .parent .child
    margin-left: 0
```

The errors:
```
  line 2:   missing colon between property and value
  line 6:   missing colon between property and value
  line 11:  missing colon between property and value
  line 15:  missing colon between property and value
  line 19:  missing colon between property and value
```

My config:
```
{
    "blocks": false,
    "brackets": "never",
    "colons": "always",
    "colors": false,
    "commaSpace": "always",
    "commentSpace": "always",
    "cssLiteral": false,
    "depthLimit": 4,
    "duplicates": true,
    "efficient": "always",
    "extendPref": "@extend",
    "globalDupe": true,
    "indentPref": 2,
    "leadingZero": "always",
    "maxErrors": false,
    "maxWarnings": false,
    "mixed": false,
    "namingConvention": false,
    "namingConventionStrict": false,
    "none": "always",
    "parenSpace": "never",
    "placeholders": false,
    "prefixVarsWithDollar": "always",
    "reporter": "stylint-stylish",
    "quotePref": false,
    "stackedProperties": "never",
    "semicolons": "never",
    "sortOrder": false,
    "universal": "never",
    "valid": true,
    "whitespace": true,
    "zeroUnits": false,
    "zIndexNormalize": false
}
```

**Comment by langdonx** [comment]
Having the same issue..  the issue exists with `brackets: "always"` as well...

    Warning: always use brackets when defining selectors
    File: index.styl
    Line: 52: @extend $anchor;

For:

    a {
        @extend $anchor;
    }

**Comment by rossPatton** [comment]
Are you both using the latest version?

**Comment by langdonx** [comment]
Yes, installed it yesterday.

    Stylint version: 1.1.3

**Comment by futekov** [comment]
Yup, logged the bug when it was `1.1.1`, it's still reproducable on `1.1.3`.

**Comment by rossPatton** [comment]
k just checking. i'll take a look this weekend.

**Comment by rossPatton** [comment]
@futekov @langdonx Can you guys let me know if you still see this issue in 1.2.0?

**Comment by langdonx** [comment]
Yes, fixed in 1.2.2.  Getting a different error now, but will file a separate issue.

**Comment by futekov** [comment]
I can confirm that since 1.2.2 the issue is fixed.

**Comment by Dylan-Chapman** [comment]
I think this may still be a problem. If, for example, you have the following:
```stylus
// select all direct children of .ChildBug
.ChildBug
  > *
    padding: 4px
```
You'll get the error:
```
line 2:  missing colon between property and value
```
Can anybody confirm this? I'm using gulp-stylint, which is using stylint 1.2.3

I can create a separate issue if it's still a problem in the newest version of stylint, but I wanted to verify first.
